extension/js/common/api/email-provider/gmail/gmail.ts

```diff
@@ -277,14 +278,20 @@ export class Gmail extends EmailProviderApi implements EmailProviderInterface {
     if (format === 'full') {
       const bodies = GmailParser.findBodies(gmailMsg);
       const attachments = GmailParser.findAttachments(gmailMsg);
-      const fromTextBody = PgpArmor.clip(Buf.fromBase64UrlStr(bodies['text/plain'] || '').toUtfStr());
+      const textBody = Buf.fromBase64UrlStr(bodies['text/plain'] || '').toUtfStr();
+      const htmlBody = Xss.htmlSanitizeAndStripAllTags(Buf.fromBase64UrlStr(bodies['text/html'] || '').toUtfStr(), '\n');
+      const fromTextBody = PgpArmor.clip(textBody);
       if (fromTextBody) {
         return { armored: fromTextBody, subject, isPwdMsg };
       }
-      const fromHtmlBody = PgpArmor.clip(Xss.htmlSanitizeAndStripAllTags(Buf.fromBase64UrlStr(bodies['text/html'] || '').toUtfStr(), '\n'));
+      const fromHtmlBody = PgpArmor.clip(htmlBody);
       if (fromHtmlBody) {
         return { armored: fromHtmlBody, subject, isPwdMsg };
       }
+      if ((fromTextBody === undefined || fromHtmlBody === undefined) && attachments.length === 0) {
+        const clippedText = (textBody || htmlBody).substring((textBody || htmlBody).search('----BEGIN PGP MESSAGE----'));
```

Comment:
Not quite clear why we need to strip characters preceding `-----BEGIN PGP MESSAGE-----`.
What would happen if we have a message with the following text:
```
An OpenPGP message starts with this header:
-----BEGIN PGP MESSAGE-----

...
```
?
Also, what would/should happen with a message that reads:
```
An OpenPGP messages starts with this header:
-----BEGIN PGP MESSAGE-----

and ends with this footer
-----END PGP MESSAGE-----
```

Reply:
>Not quite clear why we need to strip characters preceding -----BEGIN PGP MESSAGE-----.
this is the propose change does:
<img width="347" alt="Screen Shot 2022-03-16 at 4 35 29 PM" src="https://user-images.githubusercontent.com/46025304/158549385-44077c3c-4b4a-4649-b11d-4e00987d5f08.png">
Without stripping the preceding chars before the pgp begin header, it will repeat the entire message in the pgpblock.

Also, does the gray border looks inappropriate?

> Also, what would/should happen with a message that reads:

It does return a generic message says : "Error: Error during parsing. This message / key probably does not conform to a valid OpenPGP format.", it also shows generic error on the android version. not sure what's best for it.

<img width="1065" alt="Screen Shot 2022-03-16 at 4 33 26 PM" src="https://user-images.githubusercontent.com/46025304/158549359-ad24f82e-a075-4dac-b9bd-306f4f17ddf4.png">

Reply:
@martgil, can you please add the message we previously discussed
```
An OpenPGP message starts with this header:
-----BEGIN PGP MESSAGE-----

example
...
into exported-messages too?
And also, test calling Gmail.extractArmoredBlock() on this message in both raw and full formats to make sure they return the same plaintext.

Reply:
yes, sir, I can add it and add a similar test to check both raw and full formats.

Reply:
sir @rrrooommmaaa - I've added the new test for the following message above and updated the code a little bit. thank you, sir!